src/Pact/Compile.hs

```diff
@@ -585,7 +585,8 @@ abstractBody :: Compile (Term Name) -> [Arg (Term Name)] -> Compile (Scope Int T
 abstractBody term args = abstractBody' args =<< bodyForm term
 
 abstractBody' :: [Arg (Term Name)] -> Term Name -> Compile (Scope Int Term Name)
-abstractBody' args body = traverse enrichDynamic $ abstract (`elemIndex` bNames) body
+abstractBody' args body =
+  (if M.null modRefArgs then pure else traverse enrichDynamic) $ abstract (`elemIndex` bNames) body
```

Comment:
This has no semantic effect?

Reply:
Consider a body containing a `DynamicName` `foo::bar`, with a bound argument in `args` `foo:module{quux}`. Modrefs + dynamic names require explicit typing so that this code can propagate the type from the arg to the dynamic ref, which is the function of `enrichDynamic`: traverse free variables in the body to find dynamic refs that reference a bound argument. 

Thus, it is not necessary if there are no modrefs in the arguments.

Reply:
Thank you for the clarification @sirlensalot!